processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -91,7 +122,12 @@ public boolean put(String lookupName, final LookupExtractorFactory lookupExtract
       if (!lookupExtractorFactory.start()) {
         throw new ISE("start method returned false for lookup [%s]", lookupName);
       }
-      return (null == lookupMap.putIfAbsent(lookupName, lookupExtractorFactory));
+
+      boolean isAdded = (null == lookupMap.putIfAbsent(lookupName, lookupExtractorFactory));
+      if (isAdded && lookupSnapshotTaker != null) {
+        lookupSnapshotTaker.takeSnapshot(getAllAsList());
```

Comment:
yes every put you have new changes, like that you make sure if the process crash we have a recent snapshot.